iroha/src/kura.rs

```diff
@@ -180,77 +180,129 @@ impl Default for Mode {
 #[derive(Debug)]
 pub struct BlockStore {
     path: PathBuf,
+    blocks_per_file: u64,
 }
 
 impl BlockStore {
     /// Initialize block storage at `path`.
     ///
     /// # Errors
     /// - Failed to create directory.
-    pub fn new(path: &Path) -> Result<BlockStore> {
+    pub fn new(path: &Path, blocks_per_file: u64) -> Result<BlockStore> {
```

Comment:
`NonZero<u64>` would suit better i think. Also you can add second constructor for defaulting `blocks_per_file`

Reply:
Good points, I somehow didn't think of NonZero. Thanks

Reply:
General intention is to use Correct-by-construction pattern. Do not validate input, use types for type validation (types which are have some properties (like `NonZero`))